File: android/app/src/main/java/at/area23/schnapslet/constenum/PLAYEROPTIONS.java
/*
 *
 * @author           Heinrich Elsigan root@darkstar.work
 * @version          V 1.6.9
 * @since            API 26 Android Oreo 8.1
 *
 */
/*
   Copyright (C) 2000 - 2023 Heinrich Elsigan root@darkstar.work

   Schnapslet java applet is free software; you can redistribute it and/or
   modify it under the terms of the GNU Library General Public License as
   published by the Free Software Foundation; either version 2 of the
   License, or (at your option) any later version.
   See the GNU Library General Public License for more details.

*/
package at.area23.schnapslet.constenum;

/**
 * PLAYEROPTIONS enum represents possible options of playing.
 *
 * @see <a href="https://github.com/heinrichelsigan/schnapslet/wiki</a>
 */
public enum PLAYEROPTIONS {

	CHANGEATOU(1),
	CLOSESGAME(2),
	SAYPAIR(4),
	PLAYSCARD(8),
	HITSCARD(16),
	ANDENOUGH(32);

	/**
	 * Enum PLAYEROPTIONS constructor
	 *  must have private or package scope. You can not use the public access modifier.
	 */
	PLAYEROPTIONS(int value) {
		this.value = value;
	}

	private final int value;

	/**
	 * getValue()
	 * @return int value
	 */
	public int getValue() {
		return value;
	}
}



	

File: asp.net/SchnapsNet/ConstEum/PLAYEROPTIONS.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;

namespace SchnapsNet.ConstEnum
{
    [Serializable]
    public enum PLAYEROPTIONS
    {
        NONE        = 0,
        CHANGEATOU  = 1,
        SAYPAIR     = 2,        
        PLAYSCARD   = 4,
        HITSCARD    = 8,        
        CLOSESGAME  = 16,
        ANDENOUGH   = 32,
        LASTSTITCH  = 64
    }
    
    public static class PLAYEROPTIONS_Extensions
    {
        public static int GetValue(this PLAYEROPTIONS playOpt)
        {
            switch(playOpt)
            {
                case PLAYEROPTIONS.CHANGEATOU:  return 1;
                case PLAYEROPTIONS.SAYPAIR:     return 2;
                case PLAYEROPTIONS.PLAYSCARD:   return 4;
                case PLAYEROPTIONS.HITSCARD:    return 8;
                case PLAYEROPTIONS.CLOSESGAME:  return 16;
                case PLAYEROPTIONS.ANDENOUGH:   return 32;
                case PLAYEROPTIONS.LASTSTITCH:  return 64;
            }
            return 0;
        }
    }
}